content/custom-decorators.md

```diff
@@ -150,3 +150,32 @@ async findOne(user) {
   console.log(user);
 }
 ```
+
+#### Decorators composition
+
```

Comment:
#### Decorator composition

Nest provides a helper method to compose multiple decorators.  For example, suppose you want to combine all decorators related to authentication into a single decorator.  This could be done with the following construction:

```typescript
import { applyDecorators, NestCustomDecorator } from "@nestjs/core";
export function Auth(roles: UserRole | UserRole[]): NestCustomDecorator {
  return applyDecorators(
    SetMetadata("roles", Array.isArray(roles) ? roles : [roles]),
    UseGuards(AuthGuard),
    ApiBearerAuth(),
    ApiUnauthorizedResponse({ description: "Unauthorized" })
  );
}
```

You can then use this custom `@Auth()`  decorator as follows:

```typescript
import { Auth } from '../auth/auth.decorator';
@Get('')
@Auth('admin')
myAuthMethod() { [...] }
```

This has the effect of applying all four decorators with a single declaration.